lightning-invoice/src/utils.rs

```diff
@@ -129,12 +129,13 @@ where
 pub struct DefaultRouter<G: Deref<Target = NetworkGraph>, L: Deref> where L::Target: Logger {
 	network_graph: G,
 	logger: L,
+	random_bytes: [u8; 32],
```

Comment:
Thanks for having a look at this! No, very likely we don't want to re-use the randomness. However, I also found it pretty messy to expose the `random_bytes`  to through every part of the higher-up lightning-invoice API to force a user to supply a fresh value every time. Maybe instead of taking `random_bytes: [u8: 32]`, `find_route` could also take a reference to a stateful `prng: ChaCha20`. This would ensure we'd always use different randomness and could also be used to implement a 'real' shuffle in https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/routing/router.rs#L1418.

Reply:
Alternatively, `DefaultRouter` could store a `ChaCha20` in a mutex and get fresh randomness for each call, no?

Reply:
Yes, this of course also works, but we'll then still need to initiate another `ChaCha20` in `get_route` itself. It probably won't hurt though, so I'll go that route.

Reply:
Addressed in https://github.com/lightningdevkit/rust-lightning/commit/ea0ad9d079ba51e5cf6f780999661a1b7673351b.